bbq/vm/vm.go

```diff
@@ -89,7 +89,15 @@ func NewVM(
 	}
 
 	// Delegate the function invocations to the vm.
-	context.invokeFunction = vm.invokeExternally
+	context.invokeFunction = func(function Value, arguments []Value) (Value, error) {
+		// invokeExternally runs the VM, which is incorrect for native functions.
+		if function, ok := function.(NativeFunctionValue); ok {
+			result := function.Function(vm.context, nil, arguments...)
+			return result, nil
+		}
+
+		return vm.invokeExternally(function, arguments)
```

Comment:
Good catch!
We might have to move this to the `invokeExternally` function, because it also handles the case where native function is wrapped with a bound-function-pointer.
i.e: `vm.run()` only if the actual function is a non-native function

Reply:
Do you maybe have an example for when that would be the case?

Reply:
Though we might also need to fix `invokeExternally`, it might be a good idea to keep this shortcut for native functions here, as it avoids a push and pop from the stack.

Reply:
Added an item to #3804